license.go

```diff
@@ -58,20 +58,66 @@ func licenseType(name string) Type {
 	return Other
 }
 
+// A phrase is a sequence of words used as a key for startIndexes.
+// Empirically, two words are best; more is slower.
+type phrase [2]int32
+
 type license struct {
 	typ          Type
 	name         string
+	text         string
 	doc          *document
-	startIndexes map[string][]int
+	startIndexes map[phrase][]int
 }
 
 type document struct {
-	text    []byte   // Original text.
-	words   []string // Normalized words.
-	byteOff []int32  // ith byteOff is byte offset of ith word in original text.
+	text    []byte  // Original text.
+	words   []int32 // Normalized words (indexes into c.words)
+	byteOff []int32 // ith byteOff is byte offset of ith word in original text.
+}
+
+// A Checker matches a set of known licenses.
+type Checker struct {
+	licenses []license
+	urls     map[string]string
+	dict     map[string]int32 // dict maps word to index in words
+	words    []string         // list of known words
+}
+
+// A License describes a single license that can be recognized.
+// At least one of the Text or the URL should be set.
+type License struct {
+	Name string
+	Text string
+	URL  string
+}
+
+// New returns a new Checker that recognizes the given list of licenses.
+func New(licenses []License) *Checker {
```

Comment:
While this is sufficient, it would be convenient to be able to augment the builtin list of licenses; otherwise I need to find that list in the licensecheck source and copy it, and then I won't get updates.

How about exporting the list of licenses that is now assigned to the local `files` in gen_data.go?

Reply:
I don't want to export the built-in list. You should use the set _you_ need and thereby end up isolated from any changes to the built-in list.

Reply:
I thought some more about this and I think you're right, especially if the set here grows to be useful enough that most people only need one or two special cases. Added func Default.